cmd/bosun/search/search.go

```diff
@@ -284,6 +284,20 @@ func (s *Search) TagKeysByMetric(Metric string) []string {
 	return r
 }
 
+func (s *Search) TagKeysByMetricTree() map[string][]string {
+	metricToKeys := make(map[string][]string)
+	searchResult := s.Read.Tagk.Copy()
+	for metric, v := range searchResult {
+		metricToKeys[metric] = make([]string, len(v))
+		i := 0
+		for tK := range v {
+			metricToKeys[metric][i] = tK
```

Comment:
For readability I recommend making and filling the slice, and then putting it into `metricsToKeys` at the end of the loop. That may reduce the chained map/slice indexing.